dotnet/src/Extensions/Reliability.Polly/Config/DefaultHttpRetryHandler.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+namespace Microsoft.SemanticKernel.Reliability.Polly.Config;
+
+/// <summary>
+/// A basic example of a retry mechanism that retries three times with backoff.
+/// </summary>
+public class DefaultHttpRetryHandler : DelegatingHandler
+{
+    private readonly AsyncPolicy<HttpResponseMessage> _policy;
+    private readonly ILogger<DefaultHttpRetryHandler>? _logger;
+    private readonly HttpRetryConfig _config;
+    private ITimeProvider _timeProvider;
+
+    public DefaultHttpRetryHandler(HttpRetryConfig? config = null, ILoggerFactory? loggerFactory = null) : this(config, null, loggerFactory)
+    {
+    }
+
+    internal DefaultHttpRetryHandler(HttpRetryConfig? config, ITimeProvider? timeProvider, ILoggerFactory? loggerFactory)
+    {
+        this._logger = loggerFactory?.CreateLogger<DefaultHttpRetryHandler>();
+        this._config = config ?? new();
+        this._policy = this.GetPolicy();
+        this._timeProvider = timeProvider ?? new DefaultTimeProvider();
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        return await this._policy.ExecuteAsync(async () =>
```

Comment:
Consider passing cancellation token as an argument to the `ExecuteAsync` method.